src/Cli.ts

```diff
@@ -97,31 +97,44 @@ export enum ComparisonResultType {
     ERROR = "ERROR"
 }
 
-export class IonComparisonReport {
+/** comparison result with event index and message **/
+export class ComparisonResult {
+    message: string;
     result: ComparisonResultType;
+    actualIndex: number;
+    expectedIndex: number;
```

Comment:
Out of curiosity, why are you choosing to store the indices rather than the ComparisonContexts?

Reply:
Because otherwise I have to pass the entire `ComparisonContext` object to `IonEvent` which is a single event from a `ComparisonContext`'s event stream and there's no way to get the particular event from event stream except through index. So I thought to just pass the index rather than passing the object. Does this answer your concern?